samples/Kconfig

```diff
@@ -16,15 +16,54 @@ config NCS_SAMPLES_DEFAULTS
 	help
 	  Use the default configuration for NCS samples.
 
-config NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE
+if SOC_SERIES_NRF53X
+
+config AUTO_INCLUDE_CHILD_IMAGE
+	bool "Automatically add a child image when relevant"
+	select PARTITION_MANAGER_ENABLED
+	help
+	  Add a child image to the other core. If the current sample is built
+	  for the APP core the child image will be build for the NET core and
+	  vice versa.
```

Comment:
I see this is actually the case:
https://github.com/nrfconnect/sdk-nrf/blob/75c9842d6a8877448dcd89434fbbcf9af548c23e/samples/CMakeLists.txt#L103-L107

but is that sane behavior ?

If a user tries to build `samples/bluetooth/peripheral_lbs` for the nRF5340 netcore, then we'll be adding the `HCI RPMsg` to the appcore which doesn't have a radio ?
Sounds wrong to me.

Afaik, it's actually possible to run simple bluetooth samples directly on the netcore, without powering up the app core.
This is what the empty app core is used for.
In other scenarios than the specific empty app core we should probably not start adding child images to the appcore when building for the netcore.

Maybe we should only support this child image selection for the `app` --> `netcore` domain in this Kconfig part.
And then a specific use-case when building for the `netcore` where the empty app core is added as separate code part in Kconfig and CMake flow.

Reply:
I don't think you got the point, in this example i'm building for the board `nrf5340dk_nrf5340_cpunet`, but I can choose any child image for the appcore:
![image](https://user-images.githubusercontent.com/9802583/137866733-e6357574-fc5b-4f37-b6ce-8694104935fe.png)

So this means the dependencies are wrong.

For example:
- empty app code shall only be selectable when building for netcore
- HCI RPMsg child image shall only be selectable when `BT_RPMSG=y`
- nRF IEEE 802.15.4 Serialization shall only be selectable when `NRF_802154_SER_HOST=y`

A user shouldn't be able to select `HCI RPMsg` together with `NRF_802154_SER_HOST` as far as I can tell.

Please check up on the dependencies between sample configuration settings and which child images that should be available.

Also consider if the actual child image should be promptless, as I mention here: https://github.com/nrfconnect/sdk-nrf/pull/5052#discussion_r731585218